Userland/Services/FileSystemAccessServer/FileSystemAccessServer.ipc

```diff
@@ -0,0 +1,10 @@
+endpoint FileSystemAccessServer
+{
```

Comment:
Maybe the client would want to know if a request succeeded.
For `request_file_{read,write}` this could fail in the following ways:
* User declines permission.
* File system permissions deny permission.
* Any other way that the open syscall could fail.

For `prompt_file_{read,write}` this could also mean that the user cancelled the file picker dialog. For `download_file` there are also all the failures that RequestServer can have.

On another note, what is the `i32 a` for, and could it maybe have a more descriptive name?

Reply:
`i32 a` is there as a helpful friend for `IPC::File`.

When I tried to return only `IPC::File` it triggered `AK_MAKE_NONCOPYABLE` and caused the compilation to fail.

```
../.././AK/Optional.h: In instantiation of 'AK::Optional<T>::Optional(const AK::Optional<T>&) [with T = IPC::File]':
Userland/Services/FileSystemAccessServer/FileSystemAccessServerEndpoint.h:265:115:   required from 'AK::Optional<IPC::File> FileSystemAccessServerProxy<LocalEndpoint, PeerEndpoint>::request_file_read(AK::String) [with LocalEndpoint = FileSystemAccessClientEndpoint; PeerEndpoint = FileSystemAccessServerEndpoint]'
../../Userland/Applications/Browser/Tab.cpp:91:34:   required from here
../.././AK/Optional.h:55:13: error: use of deleted function 'IPC::File::File(const IPC::File&)'
   55 |             new (&m_storage) T(other.value());
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../.././AK/RefCounted.h:12,
                 from ../.././AK/StringImpl.h:10,
                 from ../.././AK/String.h:14,
                 from ../.././AK/URL.h:10,
                 from ../../Userland/Applications/Browser/History.h:9,
                 from ../../Userland/Applications/Browser/Tab.h:9,
                 from ../../Userland/Applications/Browser/Tab.cpp:8:
../../Userland/Libraries/LibIPC/File.h:17:25: note: declared here
   17 |     AK_MAKE_NONCOPYABLE(File);
      |           
```